packages/flame_spine/lib/src/skeleton_render.dart

```diff
@@ -0,0 +1,172 @@
+
+  set alignment(AlignmentGeometry value) {
+    _renderObject.alignment = value as Alignment;
+  }
+
+  BoxFit get boxFit => _renderObject.fit;
+
+  set boxFit(BoxFit value) {
+    _renderObject.fit = value;
+  }
+
+  PlayState get playState => _renderObject.playState;
+
+  set playState(PlayState value) {
+    _renderObject.playState = value;
+  }
+
+  bool get debugRendering => _renderObject.debugRendering;
+
+  set debugRendering(bool value) {
+    _renderObject.debugRendering = value;
+  }
+
+  bool get triangleRendering => _renderObject.triangleRendering;
+
+  set triangleRendering(bool value) {
+    _renderObject.triangleRendering = value;
+  }
+
+  /// Нow many percent increase the size of the animation
+  /// relative to the size of the first frame.
+  double get frameSizeMultiplier => _renderObject.frameSizeMultiplier;
+
+  set frameSizeMultiplier(double value) {
+    _renderObject.frameSizeMultiplier = value;
+  }
+
+  /// A start animation. We will use it for calculate bounds by frames.
+  String? get animation => _renderObject.animation;
+
+  set animation(String? value) {
+    _renderObject.animation = value;
+  }
+
+  void render(Canvas canvas, Size size) {
+    final bound = _renderObject.bounds;
+
+    if (bound == null) {
+      throw 'SkeletonRender was rendered but Bounds are not present ';
+    }
+
+    _paint(canvas, bound, size);
+  }
+
+  void destroy() {
+    _renderObject.dispose();
+  }
+
+  void _paint(Canvas canvas, Bounds bounds, Size size) {
+    final contentHeight = bounds.size.y;
+    final contentWidth = bounds.size.x;
+    final x = -bounds.offset.x -
+        contentWidth / 2.0 -
+        (alignment.x * contentWidth / 2.0);
+    final y = -bounds.offset.y -
+        contentHeight / 2.0 +
+        (alignment.y * contentHeight / 2.0);
+
+    var scaleX = 1.0, scaleY = 1.0;
+
+    switch (boxFit) {
+      case BoxFit.fill:
+        scaleX = size.width / contentWidth;
+        scaleY = size.height / contentHeight;
+        break;
+      case BoxFit.contain:
+        final double minScale =
+            math.min(size.width / contentWidth, size.height / contentHeight);
+        scaleX = scaleY = minScale;
+        break;
+      case BoxFit.cover:
+        final double maxScale =
+            math.max(size.width / contentWidth, size.height / contentHeight);
+        scaleX = scaleY = maxScale;
+        break;
+      case BoxFit.fitHeight:
+        final minScale = size.height / contentHeight;
+        scaleX = scaleY = minScale;
+        break;
+      case BoxFit.fitWidth:
+        final minScale = size.width / contentWidth;
+        scaleX = scaleY = minScale;
+        break;
+      case BoxFit.none:
+
+      case BoxFit.scaleDown:
+        final double minScale =
+            math.min(size.width / contentWidth, size.height / contentHeight);
+        scaleX = scaleY = minScale < 1.0 ? minScale : 1.0;
+        break;
+
+      default:
+        scaleX = scaleY = 1.0;
+        break;
+    }
```

Comment:
we have the exact same code on rive component:
https://github.com/flame-engine/flame/blob/47970224f8c9c90718c54301ee69d9cddcced87b/packages/flame_rive/lib/src/rive_component.dart#L105

we should extract this to flame core as a BoxFit extension that returns a "scale" vector2